cmd/sanitize-prow-jobs/main.go

```diff
@@ -6,36 +6,134 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"regexp"
 	"strings"
 	"sync"
 
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	prowconfig "k8s.io/test-infra/prow/config"
 	"sigs.k8s.io/yaml"
 )
 
-const defaultCluster = "api.ci"
-
 type options struct {
 	prowJobConfigDir string
+	configPath       string
 
 	help bool
 }
 
+// Config defines the structure of configuration file for this tool
+type Config struct {
+	Clusters Clusters
+}
+
+// Clusters defines the cluster parameter for each Prow job, i.e., where it runs
+type Clusters struct {
+	// the cluster context name if no other condition matches
+	Default string
+
+	Groups []Group
+	Raw    []UnnamedGroup
+	Files  []File
+
+	FileREMap map[string]*regexp.Regexp `json:"-"`
+}
+
+// Group defines a name for a group of jobs and where they run
+type Group struct {
+	// the name of the group
+	Name string
+	// the job names of the group
+	UnnamedGroup `json:",inline"`
+}
+
+// UnnamedGroup defines where a group of jobs run
+type UnnamedGroup struct {
+	// the job names
+	Jobs []string
+	// the cluster where the jobs run
+	Cluster string
+}
+
+// File defines where the jobs from a set of files run
+type File struct {
+	// the regular expression of the paths of job files
+	Paths []string
+	// the cluster where the jobs in the files run
+	Cluster string
+}
+
+func (config *Config) getClusterForJob(jobName string, path string) string {
+	for _, g := range config.Clusters.Raw {
+		for _, job := range g.Jobs {
+			if jobName == job {
+				return g.Cluster
+			}
+		}
+	}
+	for _, file := range config.Clusters.Files {
+		for _, p := range file.Paths {
+			if config.Clusters.FileREMap[p].MatchString(path) {
```

Comment:
co-locate the regex and the struct that created it

```
if file.regex.MatchString()
```

Reply:
Done.